src/services/services.ts

```diff
@@ -343,8 +345,8 @@ function createNode<TKind extends SyntaxKind>(kind: TKind, pos: number, end: num
     return node;
 }
 
-class NodeObject implements Node {
-    public kind: SyntaxKind;
+class NodeObject<TKind extends SyntaxKind> implements Node {

```

Comment:
Rather than copy & pasting the constructor ordering, can this not just `extend (objectAllocator.getNodeConstructor())`? This file only overrides the allocator with the services one at the end, so calling it when we construct these classes should get the base ones.

Reply:
I agree with the sentiment, but I'd be happy to leave it as it is in this PR until we can move all of this to compiler/ and do away with `ObjectAllocator` entirely.

Reply:
I just tried it and I think it'd work at runtime, but at compile time it gets really awkward because:

- Services adds methods to `Node`, but since `getNodeConstructor()` gives a constructor for `Node`, `NodeObject` needs overrides for everything
- The type arguments on the services constructors conflict with the compiler's allocators, so that all has to go away again
- We can't do the same for Token/Identifier/PrivateIdentifier, as they share methods provided by `TokenOrIdentifierObject` (and they're also different than NodeObject).

So, I'm sort of inclined to leave this "as-is" for the quick win until we can drop the service altogether...